### PR TITLE
add a http flatfile collector to gather key=value metrics from http sources

### PR DESCRIPTION
**Pull request by psych0d0g**
Please have a look and merge if everything is okay

### CODE REVIEW

**Comment by mattrobenolt** [comment]
What's the use case for this?

Either way, the tests you added are failing, so check into that.

Lastly, please revert the change to the `debian/changelog` file.

File: src/collectors/httpflat/httpflat.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    def _flat_to_flat_metrics(self, prefix, data):
+        data = dict(line.split("=") for line in data.split("\n"))
+        for key in data:
+            yield key, data[key]
+
+    def collect(self):
+        url = self.config['url']
+
+        req = urllib2.Request(url)
+        req.add_header('Content-type', 'application/text')
```
Comment (mattrobenolt) [inline src/collectors/httpflat/httpflat.py]: Is this required for some reason? If not, this is unnecessary.

File: src/collectors/httpflat/httpflat.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    def get_default_config(self):
+        default_config = super(HTTPFLATCollector, self).get_default_config()
+        default_config.update({
+            'path': 'httpflat',
+            'url': 'http://localhost/stat'
+        })
+        return default_config
+
+    def _flat_to_flat_metrics(self, prefix, data):
```
Comment (mattrobenolt) [inline src/collectors/httpflat/httpflat.py]: This whole thing should just be done as a generator. yielding for each line. Right now, you're splitting the lines upfront, then yielding all the parsed values. Would be better to iterate each line and parse separately.

File: src/collectors/httpflat/httpflat.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            content = resp.read()
+
+            try:
+                data = content[1:]
+            except ValueError as e:
+                self.log.error("Can't parse JSON object from %s. %s", url, e)
+            else:
+                for metric_name, metric_value in self._flat_to_flat_metrics(
+                        "", data):
+                    self.publish('voip.rtstatng.' + metric_name, metric_value)
```
Comment (mattrobenolt) [inline src/collectors/httpflat/httpflat.py]: Seems like this prefix is hardcoded? Is this all for a voip service?

**Comment by psych0d0g** [comment]
Closed for now, will redo pull request more cleanly soon

**Comment by mattrobenolt** [comment]
You didn't need to close the pull request to clean up and address feedback. :)

**Comment by psych0d0g** [comment]
i have some more stuff ill rewrite and add thats why i closed it and im going to open a new "clean" one soon

File: src/collectors/httpflat/httpflat.py (hunk 1 above)
Comment (mattrobenolt) [inline src/collectors/httpflat/httpflat.py]: Also worth noting, there is no such thing as an `application/text` type. If anything, it'd be `text/plain`, but on a request, this describes the content-type of the body. In this case, you're making a GET, so there is no body, and this isn't needed.

**Pull request closed without merging**